vignettes/function-flow.Rmd

```diff
@@ -97,113 +97,318 @@ library(dplyr)
 
-<!-- TODO: Add details on how this filtering should be done -->
+#### GLD purchases
 
-#### Glucose-lowering brand drugs for weight loss
+The function `include_gld_purchases()` uses `lmdb` to extract the dates
+of all GLD purchases.
+
+These dates are extracted by including all values beginning with "A10"
+in the `atc` variable of the `lmdb` register, except for
+glucose-lowering drugs that may be used for other conditions than
+diabetes: GLP-RAs (`atc` start with "A10BJ") or
+dapagliflozin/empagliflozin (`atc` = "A10BK01" or "A10BK03").
+
+Since the diagnosis code data on pregnancies (see below) is insufficient
+to perform censoring prior to 1997, `include_gld_purchases()` only
+extracts dates from 1997 onward by default (if Medical Birth Register
+data is available to use for censoring, the extraction window can be
+extended).
+
+This function outputs a long `data.frame` (since all dates of purchases
+must be kept for later use in classifyin diabetes type) with the
+following variables needed later in the classification part of the
+function flow:
+
+-   `pnr`: identifier variable
+-   `date`: dates of all purchases of GLD (renamed from `eksd`)
+-   `atc`: type of drug
+-   `contained_doses`: amount purchased, in number of defined daily
+    doses (DDD). Calculated as `volume` (doses contained in the
+    purchased package) times `apk` (number of packages purchased)
+-   `indication_code`: indication code of the prescription (renamed from
+    `indo`)
+
+These events are then passed to a chain of exclusion functions:
+`exclude_potential_pcos()` and `exclude_pregnancy()` described in the
+sections below.
 
-The function `exclude_wld_purchases()` uses lmdb as input and excludes
-the brand drugs Saxenda and Wegovy.
+### Exclusion events
 
-<!-- TODO: Add details on how this filtering should be done -->
+#### Metformin purchases potentially for the treatment of polycystic ovary syndrome
 
-#### Metformin purchases for women below age 40
+The function `exclude_potential_pcos()` takes the output from
+`include_gld_purchases()` and `bef` (information on sex and date of
+birth) as inputs and censors (filters out) all purchases of metformin in
+women below age 40 at the date of purchase (`atc` = "A10BA02" & `sex` =
+"woman" & age at purchase (`date`-`date_of_birth`) \< 40 years) or an
+indication code suggesting the prescription was made for treatment of
+polycystic ovary syndrome (`atc` = "A10BA02" & `sex` = "woman" &
+`indication_code` either of "0000092", "0000276" or "0000781").
 
-The function `exclude_potential_pcos()` as input to exclude all
-purchases of metformin by women below age 40 (i.e., \<= 39 years old) at
-the date of purchase. It relies on `bef` as input.
+This function only performs a filtering operation, and output retains
+the same structure and variables as the input passed from
+`include_gld_purchases()`. After these exclusions are made, the output
+is passed to `exclude_pregnancy()` for further censoring, described
```

Comment:
The weight-loss drugs are no longer an issue, since we stopped using GLP1-RA drugs for inclusion (all weightloss drugs are GLP1-RAs). They're filtered out in the very beginning of include_gld_purchases() (lines 274-278).